tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -3402,15 +3402,43 @@ SpirvEmitter::processACSBufferAppendConsume(const CXXMemberCallExpr *expr) {
   if (isAppend) {
     // Write out the value
     auto *arg0 = doExpr(expr->getArg(0));
+    if (!arg0)
+      return nullptr;
+
     if (!arg0->isRValue()) {
-      arg0 = spvBuilder.createLoad(bufferElemTy, arg0);
+      arg0 = spvBuilder.createLoad(arg0->getAstResultType() == QualType()
```

Comment:
```
arg0->getAstResultType() == QualType()
```
means `arg0` does not have an AST QualType. This should never be the case... 
Is there a case where arg0 doesn't have a QualType? what case?

Reply:
When `arg0` is a pointer, `arg0->getAstResultType() == QualType()` is true.
I am not sure there is a case that `arg0` is a pointer after updating `Consume()` part as below, but this branch will be executed if `!arg0->isRValue()` is true which means `arg0` can be a pointer.